### PR TITLE
Added JUnit tests for exception and model packages

### PR DESCRIPTION
**Pull request by sonja-ct**
This PR JUnit tests for classes in exception and model packages in rest-api/resources module:

 - exception package:
RestApiRuntimeException class
SessionNotPopulatedException class

- model package
ByteArrayParam class
CountResult class
DateParam class
EntityId class
MetricType class
ScopeId class
StorableEntityId class

Also it contains JUnit tests for class AbstractKapuaResource.

Signed-off-by: Sonja <sonja.matic@endava.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]
hey @Coduz,
I have checked this PR and it looks OK. This is part of "rest api unit tests" that we have been talking about (achieving code coverage for rest-api module through unit tests).
After all these PRs are merged and we have framework for Rest Api testing in place, we will add high-level/integration rest api tests, that will check the actual fucntionality.
Please take a look and merge when you can.

**Review by LeoNerdoG — APPROVED** [review]

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    public void restApiRuntimeExceptionNullCodeTest() {
+        RestApiRuntimeException restApiRuntimeException = new RestApiRuntimeException(null);
+
+        assertNull("Null expected.", restApiRuntimeException.getCode());
+        assertNull("Null expected.", restApiRuntimeException.getCause());
+        try {
+            restApiRuntimeException.getMessage();
+            fail("NullPointerException expected.");
+        } catch (Exception e) {
+            assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    public void restApiRuntimeExceptionNullCodeArgumentsTest() {
+        RestApiRuntimeException restApiRuntimeException = new RestApiRuntimeException(null, stringObject, intObject, charObject);
+
+        assertNull("Null expected.", restApiRuntimeException.getCode());
+        assertNull("Null expected.", restApiRuntimeException.getCause());
+        try {
+            restApiRuntimeException.getMessage();
+            fail("NullPointerException expected.");
+        } catch (Exception e) {
+            assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    public void restApiRuntimeExceptionNullCodeCauseArgumentsTest() {
+        RestApiRuntimeException restApiRuntimeException = new RestApiRuntimeException(null, cause, stringObject, intObject, charObject);
+
+        assertNull("Null expected.", restApiRuntimeException.getCode());
+        assertEquals("Expected and actual values should be the same.", cause, restApiRuntimeException.getCause());
+        try {
+            restApiRuntimeException.getMessage();
+            fail("NullPointerException expected.");
+        } catch (Exception e) {
+            assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+@Category(JUnitTests.class)
+public class DateParamTest extends Assert {
+
+    @Test
+    public void dateParamTest() throws KapuaIllegalArgumentException {
+        String[] dates = {"2020-10-27T21:32:52", " 2020-10-27T21:32:52+02:00", " 2020-10-27T19:32:52Z", " 2020-10-27T19:32:52+00:00"};
+
+        for (int i = 0; i < dates.length; i++) {
+            DateParam dateParam = new DateParam(dates[i]);
+            Assert.assertTrue("True expected", dateParam.getDate() instanceof Date);
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java]: DateParam.getDate() returns a Date.
So this is always true

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            Assert.assertNotNull("Null not expected.", dateParam.getDate());
+        }
+    }
+
+    @Test
+    public void dateParamNullTest() throws KapuaIllegalArgumentException {
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java:35]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        try {
+            dateParam.getDate();
+            Assert.fail("NullPointerException expected.");
+        } catch (Exception e) {
+            Assert.assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+        }
+    }
+
+    @Test(expected = KapuaIllegalArgumentException.class)
+    public void dateParamInvalidDateTest() throws KapuaIllegalArgumentException {
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java:42]: This is the correct way of expecting exception in a test! 😁

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/EntityIdTest.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        assertEquals("Expected and actual values should be the same.", "1", entityId.getId().toString());
+        assertEquals("Expected and actual values should be the same.", "1", entityId.toString());
+
+        entityId.setId(null);
+        Assert.assertNull("Null expected.", entityId.getId());
+        try {
+            entityId.toString();
+            Assert.fail("NullPointerException expected.");
+        } catch (Exception e) {
+            Assert.assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/EntityIdTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/ScopeIdTest.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        assertEquals("Expected and actual values should be the same.", BigInteger.TEN, scopeId.getId());
+        assertEquals("Expected and actual values should be the same.", "10", scopeId.toString());
+
+        scopeId.setId(null);
+        Assert.assertNull("Null expected.", scopeId.getId());
+        try {
+            scopeId.toString();
+            Assert.fail("NullPointerException expected.");
+        } catch (Exception e) {
+            Assert.assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/ScopeIdTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

**Review by Coduz — CHANGES_REQUESTED** [review]

**Review by LeoNerdoG — APPROVED** [review]

**Comment by sonja-ct** [comment]
Hey @Coduz I made requested changes, so you can merge this when you have time. Thanks

**Comment by sonja-ct** [comment]
Hey @Coduz I made requested changes, so you can merge this when you have time. Thanks

**Pull request merged**